From 5d539631c3a758f23282a3de8de3e6da4d7d4dc2 Mon Sep 17 00:00:00 2001
From: naoto <nnomura98@gmail.com>
Date: Tue, 11 Oct 2022 15:14:46 +0200
Subject: [PATCH 08/10] no options

Added option to disable stdin, so can be used as input only
---
 dmenu.1 | 3 +++
 dmenu.c | 7 +++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/dmenu.1 b/dmenu.1
index 17533ad..acf5898 100644
--- a/dmenu.1
+++ b/dmenu.1
@@ -5,6 +5,7 @@ dmenu \- dynamic menu
 .B dmenu
 .RB [ \-bfivP ]
 .RB [ \-lbp ]
+.RB [ \-no ]
 .RB [ \-l
 .IR lines ]
 .RB [ \-h
@@ -64,6 +65,8 @@ dmenu matches menu items case insensitively.
 .B \-lbp
 dmenu displays matching lines below prompt
 .TP
+.B \-no
+dmenu will ignore stdin, keyboard input only
 .B \-P
 dmenu will not directly display the keyboard input, but instead replace it with dots. All data from stdin will be ignored.
 .BI \-Po " offset"
diff --git a/dmenu.c b/dmenu.c
index ca46812..365c668 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -54,6 +54,7 @@ static struct item *prev, *curr, *next, *sel;
 static int mon = -1, screen;
 static int lines_below_prompt = 0;
 static int password_top_offset = 0;
+static int no_options = 0;
 
 static Atom clip, utf8;
 static Display *dpy;
@@ -662,7 +663,7 @@ readstdin(void)
 	size_t i, junk, size = 0;
 	ssize_t len;
 
-	if(passwd) {
+	if(passwd || no_options) {
 		lines = 0;
 		return;
 	}
@@ -840,7 +841,7 @@ usage(void)
 {
 	die("usage: dmenu [-bfivP] [-l lines] [-h height] [-p prompt] [-fn font] [-m monitor]\n"
 	    "             [-x xoffset] [-y yoffset] [-z width]\n"
-	    "             [-lbp]\n"
+	    "             [-lbp] [-no]\n"
 	    "             [-nb color] [-nf color] [-sb color] [-sf color] [-w windowid]\n"
 	    "             [-d separator] [-D separator]");
 }
@@ -869,6 +870,8 @@ main(int argc, char *argv[])
 			fstrstr = cistrstr;
 		} else if (!strcmp(argv[i], "-P"))   /* is the input a password */
 			passwd = 1;
+		else if (!strcmp(argv[i], "-no"))  /* disable showing options (input only) */
+			no_options = 1;
 		else if (i + 1 == argc)
 			usage();
 		/* these options take one argument */
-- 
2.38.0

